### PR TITLE
Fixed #236

### LINKED ISSUE #236: checkpoint_step should be made functional again
In the run from last night we ran into the problem of issue #239, the batch size for the validation sets needs to be smaller than for the training set or an OOM is thrown.

However, in the current code one can not decouple checkpointing and validation runs. Thus one can not checkpoint without validating which currently causes a OOM error.

To partially solve this checkpointing should be decoupled from validating. Validating should checkpoint, but checkpointing should not imply validating.

### PR DESCRIPTION
**Pull request by kdavis-mozilla**
Quick fix

### CODE REVIEW

File: DeepSpeech.ipynb [hunk 1]
```diff
@@ -1655,7 +1676,13 @@
...
     "        if result is not None:\n",
     "            train_wer = result\n",
-    "            \n",
+    "        \n",
+    "        # Checkpoint step\n",
+    "        if is_checkpoint_step:\n",
+    "            print \"Hibernating training session into directory\", \"%s\" % checkpoint_dir\n",
+    "            checkpoint_path = os.path.join(checkpoint_dir, 'model.ckpt')\n",
+    "            # Saving session's model into checkpoint path\n",
+    "            persist_model(train_context, session, checkpoint_path, epoch)\n",
```
Comment (tilmankamp) [inline DeepSpeech.ipynb]: How about checking for `not is_validation_step` to avoid double-saving?

Comment (kdavis-mozilla) [inline DeepSpeech.ipynb]: Ok. (As a checkpoint takes about 6-7 sec on my local machine, I just got a little lazy.)

**Comment by kdavis-mozilla** [comment]
@tilmankamp Could you take another look, thanks.

**Comment by tilmankamp** [comment]
LGTM

**Pull request merged**